src/test/java/com/thebois/models/beings/BeingTests.java

```diff
@@ -445,84 +440,117 @@ public void allStructuresCompletedNewPlaced() {
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
 
-        final Random mockRandom = Mockito.mock(Random.class);
-        when(mockRandom.nextInt(anyInt())).thenReturn(0);
+        final IStructure mockStructureA = Mockito.mock(IStructure.class);
+        final IStructure mockStructureB = Mockito.mock(IStructure.class);
 
-        world.createStructure(positionA);
-        world.createStructure(positionB);
+        final Position positionA = new Position(10f, 20f);
+        final Position positionB = new Position(0f, 5f);
 
-        final IStructure structureAtA = world.getStructureAt(positionA).get();
-        final IStructure structureAtB = world.getStructureAt(positionB).get();
+        when(mockFinder.findNearestIncompleteStructure(any()))
+            .thenReturn(Optional.of(mockStructureA)).thenReturn(Optional.of(mockStructureB));
 
-        for (int i = 0; i < 9; i++) {
-            structureAtA.deliverItem(new Log());
-            structureAtA.deliverItem(new Rock());
+        when(mockStructureA.getPosition()).thenReturn(positionA);
+        when(mockStructureB.getPosition()).thenReturn(positionB);
 
-            structureAtB.deliverItem(new Log());
-            structureAtB.deliverItem(new Rock());
-        }
+        when(mockStructureA.isCompleted()).thenReturn(true);
 
-        final Pawn testPawn = new Pawn(startPosition,
-                                       new Position(),
-                                       mockRandom,
-                                       pathFinder,
-                                       world);
+        final Pawn pawn = new Pawn(new Position(),
+                                   new Position(),
+                                   new Random(),
+                                   mockPathFinder,
+                                   mockFinder);
 
         // Act
-        final float initialBuiltRatio = structureAtA.builtStatus();
-        // deliver items to structureAtA
-        testPawn.update();
-        // update to find new un-built structure
-        testPawn.update();
+        // Finds A
+        pawn.update();
+        // Discard A, find B
+        pawn.update();
+
+        // Assert
+        final int callsExpected = 2;
+        verify(mockFinder, times(callsExpected)).findNearestIncompleteStructure(any());
+    }
+
+    @Test
+    public void pawnLooksForANewPawnWhenDestinationIsNotEqualToNeighborOfClosestStructure() {
+        // Arrange
+        final IPathFinder mockPathFinder = Mockito.mock(IPathFinder.class);
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
+
+        final IStructure mockStructure = Mockito.mock(IStructure.class);
+
+        final Position structurePosition = new Position(10f, 20f);
+        final Position positionA = new Position(0f, 0f);
+        final Position positionB = new Position(1f, 1f);
+        final Position positionC = new Position(2f, 2f);
+        final Position positionD = new Position(3f, 3f);
```

Comment:
style

If these positions are not referenced anywhere, they'll be much less confusing if they're inlined into a list called `path`:
```java
List<Position> path = List.of(new Position(0f, 0f), [...]);
```
Otherwise the reader is wondering what A B C D means and what they're used for. If they have meaning outside of just being part of the path, you should try to give them more meaningful names.